### PR TITLE
Move to Github Actions from Travis

### PR DESCRIPTION
**Pull request by gaborbernat**
It's faster and better integrated with Github. 

TODO: check how to migrate the deploy phase.

### CODE REVIEW

**Comment by erikseulean** [comment]
CC: @kpfleming

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
```
Comment (kpfleming) [inline .github/workflows/pre-commit.yml:12]: Use `checkout@v2` here.

**Comment by kpfleming** [comment]
Why are there three workflows which all run on all pushes and pull requests?

**Comment by gaborbernat** [comment]
Well slightly different things need to happen for test runners, for pre commit and for other checks.

**Comment by kpfleming** [comment]
Would you mind changing the workflow name from 'pre-commit' to 'lint' or something similar? I understand that you're using the pre-commit tool to run the operations, but in Git-land 'pre-commit' means something specific, and a GitHub Action workflow can't actually be run before a commit is made :-)

**Comment by kpfleming** [comment]
I'm OK with this change in principle; once everyone is happy with the workflows this can be merged with an override for the Travis check. At that point I'll make the Travis check no longer 'required', but we can leave it enabled until we're certain that the Actions workflows have fully replaced it.

**Comment by gaborbernat** [comment]
Will do the change and see how we can migrate the release workflow.

**Comment by kpfleming** [comment]
The release workflow will require me to inject the necessary credentials into the repository as secrets. I've done that already (for a personal project), it works fine, and will be a complete replacement for storing the secrets in Travis-CI. The link below is an example of obtaining a secret for logging into Quay for pushing a container image:

https://github.com/kpfleming/ansible-pdns-auth-api/blob/master/.github/workflows/build-image.yml

**Comment by gaborbernat** [comment]
@kpfleming can you set up a secret token as describe inside https://github.com/pypa/gh-action-pypi-publish#pypi-publish-github-action?

**Comment by kpfleming** [comment]
Done. My only request is that you don't use that secret with 'untrusted' actions, as they could do anything they wanted to with the secret. An Action from github.com/pypa is fine though, we trust PyPA :-)

**Pull request merged**